pages/docs/reference/lambdas.md

```diff
@@ -430,6 +430,21 @@ print(sum)
 
 </div>
 
+For lambdas that don't capture any variables (also know as _stateless_ lambdas), there is a built-in optimization:
+the compiler creates only one instance for each such lambda. So, all instances of a stateless lambda point to the same instance.
+
+<div class="sample" markdown="1" theme="idea" data-highlight-only>
+ 
+ ```kotlin
+ val lambdaProvider = {
+    { Unit }  // returns a stateless lambda
+}
```

Comment:
The example with a lambda stored in a property and returning another Unit-returning lambda is rather complicated to illustrate the concept of stateless lambda.

An alternative could be a function returning either a stateless or stateful lambda:
```kotlin
fun searchPredicate(expected: String? = null): (String?) -> Boolean {
    if (expected == null)
        return { value -> value == null } // stateless
    else
        return { value -> value == expected } // capturing
}
```